ros_buildfarm/common.py

```diff
@@ -644,3 +646,30 @@ def get_implicitly_ignored_package_names(cached_pkgs, explicitly_ignored_pkg_nam
         break
 
     return ignored_pkg_names.difference(explicitly_ignored_pkg_names)
+
+
+def get_package_conditional_context(index, rosdistro_name):
+    python_version = index.distributions[rosdistro_name].get('python_version')
+    ros_version = {
+        'ros1': '1',
+        'ros2': '2',
+    }.get(index.distributions[rosdistro_name].get('distribution_type'))
+
+    condition_context = {
+        'ROS_DISTRO': rosdistro_name,
+    }
+
+    if python_version:
+        condition_context['ROS_PYTHON_VERSION'] = python_version
+    if ros_version:
+        condition_context['ROS_VERSION'] = ros_version
+
+    return condition_context
```

Comment:
Existing code creates this dictionary already in multiple other places. I would rather prefer to see this consolidated than added yet another case.

Reply:
Addressed in #761